### DISCUSSION TITLE
Possible Bug with mfnwt compilation using mfpymake on Ubuntu

### DISCUSSION THREAD

**Discussion by BrockOliver** [modflowpy/flopy#1831] [General]
I compiled mfnwt on my mac (Os Monterey) using pymake version 1.2.5 and successfully ran a mixed steady state/transient simulation. I generated my input files using flopy version 3.3.7. I moved this model to my Ubuntu server and attempted to run the same simulation with the same flopy version, but received the error printed at the bottom of this post. I believe this is an error with how the mfnwt .exe is being compiled on my server because everything works when I run the exact same model on my mac. Thanks for helping with this issue!

**Error:**
_Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.

Backtrace for this error:

#0  0x7FA589FCF6D7

#1  0x7FA589FCFD1E

#2  0x7FA58A30E3FF

#3  0x672998 in gwf2sfr7fm_

#4  0x643FAC in MAIN__ at MF_NWT.f:?

['', '                                  MODFLOW-NWT-SWR1 ', '    U.S. GEOLOGICAL SURVEY MODULAR FINITE-DIFFERENCE GROUNDWATER-FLOW MODEL', '                             WITH NEWTON FORMULATION', '                             Version 1.2.0 03/01/2020                        ', '                    BASED ON MODFLOW-2005 Version 1.12.0 02/03/2017                       ', '', '                    SWR1 Version 1.04.0 09/15/2016                       ', '', ' Using NAME file: sl_nwt.nam ', ' Run start date and time (yyyy/mm/dd hh:mm:ss): 2023/06/16  9:04:53', '', ' Solving:  Stress period:     1    Time step:     1    Groundwater-Flow Eqn.', '', 'Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.', '', 'Backtrace for this error:', '#0  0x7FA589FCF6D7', '#1  0x7FA589FCFD1E', '#2  0x7FA58A30E3FF', '#3  0x672998 in gwf2sfr7fm_', '#4  0x643FAC in MAIN__ at MF_NWT.f:?']

Traceback (most recent call last):

  File "modflow/pygsflow/sl_modflow/notebooks/Summary.py", line 363, in <module>

    raise Exception('MODFLOW did not terminate normally.')

Exception: MODFLOW did not terminate normally._

**Comment by wpbonelli** [comment]
Thanks for the report @BrockOliver, to attempt reproduce this the Ubuntu version would be needed as well as any model code you are able to provide

**Comment by BrockOliver** [comment]
Hi W-Bonelli,

Yes, the Linux OS version appears to be as follows:

Linux-3.10.0-1160.88.1.el7.x86_64-x86_64-with-centos-7.9.2009-Core

I added you to the repo (https://github.com/BrockOliver/sl_modflow.git), which contains all of the code and data. Let me know if the DEM isn’t there and I can send it.

Thanks for helping with this issue. I have been spinning my wheels for a while…

Brett

On Jun 27, 2023, at 2:00 PM, w-bonelli ***@***.******@***.***>> wrote:



Thanks for the report @BrockOliver<https://github.com/BrockOliver>, to attempt reproduce this the Ubuntu version would be needed as well as any model code you are able to provide

**Reply by wpbonelli** [reply]
have you tried the linux distribution here https://github.com/MODFLOW-USGS/executables? and seen the same issue you are seeing with the mfnwt you compiled with pymake on your server? the linux exe should run on a fairly recent Ubuntu. but the version string you provided looks like centos so there may be other problems as binaries are not tested on centos/rhel-based linux that I know of

**Comment by BrockOliver** [comment]
Yes, I have tried that distribution. I saw the same issue as when I compiled pymake on my server:

FloPy is using the following executable to run the model: /home/b*****/soft/modflow/pygsflow/sl_modflow/bin/mfnwt

                                  MODFLOW-NWT-SWR1
    U.S. GEOLOGICAL SURVEY MODULAR FINITE-DIFFERENCE GROUNDWATER-FLOW MODEL
                             WITH NEWTON FORMULATION
                             Version 1.3.0 07/01/2022
                    BASED ON MODFLOW-2005 Version 1.12.0 02/03/2017

                    SWR1 Version 1.05.0 03/10/2022

 Using NAME file: sl_nwt.nam
 Run start date and time (yyyy/mm/dd hh:mm:ss): 2023/06/28 10:39:52

forrtl: error (65): floating invalid
Image              PC                Routine            Line        Source
libc-2.17.so       00007F2292E9B400  Unknown               Unknown  Unknown
mfnwt              00000000006158C4  ichkstrbot_module          28  gwf2sfr7_NWT.f
mfnwt              000000000060839F  gwf2sfr7ar_               702  gwf2sfr7_NWT.f
mfnwt              0000000000732086  MAIN__                    116  MF_NWT.f
mfnwt              00000000004041DD  Unknown               Unknown  Unknown
libc-2.17.so       00007F2292E87555  __libc_start_main     Unknown  Unknown
mfnwt              00000000004040FE  Unknown               Unknown  Unknown
['', '                                  MODFLOW-NWT-SWR1 ', '    U.S. GEOLOGICAL SURVEY MODULAR FINITE-DIFFERENCE GROUNDWATER-FLOW MODEL', '                             WITH NEWTON FORMULATION', '                             Version 1.3.0 07/01/2022                        ', '                    BASED ON MODFLOW-2005 Version 1.12.0 02/03/2017                       ', '', '                    SWR1 Version 1.05.0 03/10/2022                       ', '', ' Using NAME file: sl_nwt.nam ', ' Run start date and time (yyyy/mm/dd hh:mm:ss): 2023/06/28 10:39:52', '', 'forrtl: error (65): floating invalid', 'Image              PC                Routine            Line        Source             ', 'libc-2.17.so       00007F2292E9B400  Unknown               Unknown  Unknown', 'mfnwt              00000000006158C4  ichkstrbot_module          28  gwf2sfr7_NWT.f', 'mfnwt              000000000060839F  gwf2sfr7ar_               702  gwf2sfr7_NWT.f', 'mfnwt              0000000000732086  MAIN__                    116  MF_NWT.f', 'mfnwt              00000000004041DD  Unknown               Unknown  Unknown', 'libc-2.17.so       00007F2292E87555  __libc_start_main     Unknown  Unknown', 'mfnwt              00000000004040FE  Unknown               Unknown  Unknown']
Traceback (most recent call last):
  File "/home/b******/soft/modflow/pygsflow/sl_modflow/notebooks/Summary.py", line 363, in <module>
    raise Exception('MODFLOW did not terminate normally.')
Exception: MODFLOW did not terminate normally.


Thank you,
Brett

On Jun 27, 2023, at 7:01 PM, w-bonelli ***@***.******@***.***>> wrote:



have you tried the linux distribution here https://github.com/MODFLOW-USGS/executables ? and seen the same issue you are seeing with the mfnwt you compiled with pymake on your server?

**Comment by jlarsen-usgs** [comment]
@BrockOliver 

The error message seems to indicate that there is a problem with the SFR solution. Have you opened up the list file and checked to see if there are any errors or warnings from MODFLOW within the list file? I've seen this error before when there are unrealistic parameter values in the SFR reach data (data set 2) or in the transient segment data (data set 6b) [guide to SFR package](https://water.usgs.gov/ogw/modflow-nwt/MODFLOW-NWT-Guide/sfr.html).

**Comment by BrockOliver** [comment]
Hi Joshua,

Thank you for the suggestion. I don’t know if the problem lies in the solvers getting tripped up by a bad sfr file, because I can run this exact model locally using the same floppy version as on the server. I think the real difference lies in how pymake compiled mfnwt on the server.  I tried compiling mfnwt using a different pymake version and received this error, which is slightly different from the error that I sent in my last message:


  MODFLOW-NWT-SWR1
    U.S. GEOLOGICAL SURVEY MODULAR FINITE-DIFFERENCE GROUNDWATER-FLOW MODEL
                             WITH NEWTON FORMULATION
                             Version 1.2.0 03/01/2020
                    BASED ON MODFLOW-2005 Version 1.12.0 02/03/2017

                    SWR1 Version 1.04.0 09/15/2016

 Using NAME file: sl_nwt.nam
 Run start date and time (yyyy/mm/dd hh:mm:ss): 2023/06/28 11:32:23

 Solving:  Stress period:     1    Time step:     1    Groundwater-Flow Eqn.

Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.

Backtrace for this error:
#0  0x7F26FEC166D7
#1  0x7F26FEC16D1E
#2  0x7F26FEF553FF
#3  0x640478 in gwf2sfr7fm_
#4  0x6B2A7C in MAIN__ at MF_NWT.f:?
['', '                                  MODFLOW-NWT-SWR1 ', '    U.S. GEOLOGICAL SURVEY MODULAR FINITE-DIFFERENCE GROUNDWATER-FLOW MODEL', '                             WITH NEWTON FORMULATION', '                             Version 1.2.0 03/01/2020                        ', '                    BASED ON MODFLOW-2005 Version 1.12.0 02/03/2017                       ', '', '                    SWR1 Version 1.04.0 09/15/2016                       ', '', ' Using NAME file: sl_nwt.nam ', ' Run start date and time (yyyy/mm/dd hh:mm:ss): 2023/06/28 11:32:23', '', ' Solving:  Stress period:     1    Time step:     1    Groundwater-Flow Eqn.', '', 'Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.', '', 'Backtrace for this error:', '#0  0x7F26FEC166D7', '#1  0x7F26FEC16D1E', '#2  0x7F26FEF553FF', '#3  0x640478 in gwf2sfr7fm_', '#4  0x6B2A7C in MAIN__ at MF_NWT.f:?']
Traceback (most recent call last):
  File "/home/b*****/soft/modflow/pygsflow/sl_modflow/notebooks/Summary.py", line 363, in <module>
    raise Exception('MODFLOW did not terminate normally.')
Exception: MODFLOW did not terminate normally.

Yes, the lst for the simulation mentioned in my previous email just ends as it is creating the sfr cells. However, the lst for the simulation shown above ends further into the simulation.

Thank you,
Brett

On Jun 28, 2023, at 11:24 AM, Joshua Larsen ***@***.******@***.***>> wrote:



@BrockOliver<https://github.com/BrockOliver>

The error message seems to indicate that there is a problem with the SFR solution. Have you opened up the list file and checked to see if there are any errors or warnings from MODFLOW within the list file? I've seen this error before when there are unrealistic parameter values in the SFR reach data in the reach data (data set 2) or in the transient segment data (data set 6b) guide to SFR package<https://water.usgs.gov/ogw/modflow-nwt/MODFLOW-NWT-Guide/sfr.html>.

**Comment by BrockOliver** [comment]
Hi all,

I am still working through this error, but I did find that the error arises because floppy is producing a .sfr file with Nan values (see attached sfr file). When this file is created without the Nan values the model executes normally.

I am not sure what causes some file to have Nan values and others to not. Any ideas? Again I am using floppy 3.3.6.

Cheers,
Brett



On Jun 28, 2023, at 12:15 PM, Oliver, Brett ***@***.******@***.***>> wrote:

Hi Joshua,

Thank you for the suggestion. I don’t know if the problem lies in the solvers getting tripped up by a bad sfr file, because I can run this exact model locally using the same floppy version as on the server. I think the real difference lies in how pymake compiled mfnwt on the server.  I tried compiling mfnwt using a different pymake version and received this error, which is slightly different from the error that I sent in my last message:


  MODFLOW-NWT-SWR1
    U.S. GEOLOGICAL SURVEY MODULAR FINITE-DIFFERENCE GROUNDWATER-FLOW MODEL
                             WITH NEWTON FORMULATION
                             Version 1.2.0 03/01/2020
                    BASED ON MODFLOW-2005 Version 1.12.0 02/03/2017

                    SWR1 Version 1.04.0 09/15/2016

 Using NAME file: sl_nwt.nam
 Run start date and time (yyyy/mm/dd hh:mm:ss): 2023/06/28 11:32:23

 Solving:  Stress period:     1    Time step:     1    Groundwater-Flow Eqn.

Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.

Backtrace for this error:
#0  0x7F26FEC166D7
#1  0x7F26FEC16D1E
#2  0x7F26FEF553FF
#3  0x640478 in gwf2sfr7fm_
#4  0x6B2A7C in MAIN__ at MF_NWT.f:?
['', '                                  MODFLOW-NWT-SWR1 ', '    U.S. GEOLOGICAL SURVEY MODULAR FINITE-DIFFERENCE GROUNDWATER-FLOW MODEL', '                             WITH NEWTON FORMULATION', '                             Version 1.2.0 03/01/2020                        ', '                    BASED ON MODFLOW-2005 Version 1.12.0 02/03/2017                       ', '', '                    SWR1 Version 1.04.0 09/15/2016                       ', '', ' Using NAME file: sl_nwt.nam ', ' Run start date and time (yyyy/mm/dd hh:mm:ss): 2023/06/28 11:32:23', '', ' Solving:  Stress period:     1    Time step:     1    Groundwater-Flow Eqn.', '', 'Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.', '', 'Backtrace for this error:', '#0  0x7F26FEC166D7', '#1  0x7F26FEC16D1E', '#2  0x7F26FEF553FF', '#3  0x640478 in gwf2sfr7fm_', '#4  0x6B2A7C in MAIN__ at MF_NWT.f:?']
Traceback (most recent call last):
  File "/home/b*****/soft/modflow/pygsflow/sl_modflow/notebooks/Summary.py", line 363, in <module>
    raise Exception('MODFLOW did not terminate normally.')
Exception: MODFLOW did not terminate normally.

Yes, the lst for the simulation mentioned in my previous email just ends as it is creating the sfr cells. However, the lst for the simulation shown above ends further into the simulation.

Thank you,
Brett

On Jun 28, 2023, at 11:24 AM, Joshua Larsen ***@***.******@***.***>> wrote:



@BrockOliver<https://github.com/BrockOliver>

The error message seems to indicate that there is a problem with the SFR solution. Have you opened up the list file and checked to see if there are any errors or warnings from MODFLOW within the list file? I've seen this error before when there are unrealistic parameter values in the SFR reach data in the reach data (data set 2) or in the transient segment data (data set 6b) guide to SFR package<https://water.usgs.gov/ogw/modflow-nwt/MODFLOW-NWT-Guide/sfr.html>.
